Hw6/task_1.py

```diff
@@ -0,0 +1,44 @@
+class frange:
+    def __init__(self, *args):
+        if len(args) == 1:
+            self.left = 0.0
+            self.right = args[0]
+            self.step = 1.0
+        elif len(args) == 2:
+            self.left = args[0]
+            self.right = args[1]
+            self.step = 1.0
+        elif len(args) == 3:
+            self.left = args[0]
+            self.right = args[1]
+            self.step = args[2]
+        else:
+            raise TypeError('Must be at most 3 arguments')
+        self.now = self.left
+
+    def __next__(self):
+        if (self.step > 0 and self.now >= self.right) or (self.step < 0 and self.now <= self.right):
+            raise StopIteration
+        else:
+            val = self.now
```

Comment:
use full name of words, val -> value, not now but current value. 

Here I dont understand why do you need val and self now, they are the same